drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -133,6 +133,13 @@ struct bcm2835_isp_dev {
 	int b_gain;
 };
 
+/*
+ * We want to instantiate 2 independent instances allowing 2 simultaneous users
+ * of the ISP hardware.
+ */
+#define BCM2835_ISP_NUM_INSTANCES 2
+struct bcm2835_isp_dev *bcm2835_isp_instance[BCM2835_ISP_NUM_INSTANCES];
```

Comment:
Can we devm_kzalloc this in ```bcm2835_isp_probe```, and store that parent pointer via ```platform_set_drvdata(pdev, dev);```?

I wouldn't expect to get two instances of this driver probed, but currently they'll fight over this structure. In a very ugly manner.

Reply:
Ack